addons/rose/addon/components/rose/form/checkbox/group/index.js

```diff
@@ -25,13 +24,51 @@ export default class RoseFormCheckboxGroupComponent extends Component {
     const selectedItems = this.args.selectedItems || [];
     const currentItems = [...selectedItems];
 
-    if (currentItems.includes(item)) {
-      const i = currentItems.indexOf(item);
+    if (this.doesItemExist(currentItems, item)) {
```

Comment:
Are we opposed to pulling in a package like lodash's `isEqual` to handle these equality checks? This component currently assumes the items are primitives and can't handle complex types (e.g. arrays or objects).

This would remove the need for the consumer to pass in a custom equality check (though it is more flexible in the scenario where you might want nearly similar objects to be considered equal).

Reply:
If the arrays/objects are equal by reference, then an equality test would evaluate to true.  Is there a case where we are comparing copies of complex types?

Reply:
Yeah, currently I'm using the `resource-filter` and unfortunately the `allowedValues` will be different from the `selectedValue` as it deserializes the `selectedValue` from query strings which won't be the same as the ones created from `allowedValues`